agent/api/task/task.go

```diff
@@ -2374,3 +2385,21 @@ func (task *Task) AssociationByTypeAndName(associationType, associationName stri
 
 	return nil, false
 }
+
+// GetContainerIndex returns the index of the container in the container list. This doesn't count internal container.
+func (task *Task) GetContainerIndex(containerName string) int {
+	task.lock.RLock()
+	defer task.lock.RUnlock()
+
+	idx := 0
+	for _, container := range task.Containers {
```

Comment:
you can do the same thing by doing:
for idx, contianer

Reply:
i'm trying to skip counting internal container..

Reply:
if the index actually doesn't matter (since it just needs to be unique), for loop's index can just be used.

Reply:
GetContainerIndex is invoked two times, one is during PostUnmarshalTask, the other is createContainer, and the index needs to remain the same for those two. i'm trying to avoid an assumption that no internal container is injected between these two. lmk if that doesn't make sense..